src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -84,21 +84,15 @@ static void ProcessTypeArguments (OperationAnalysisContext context, IInvocationO
 
 		static IEnumerable<Diagnostic> GetDynamicallyAccessedMembersDiagnostics (ValueSet<SingleValue> source, ValueSet<SingleValue> target, Location location)
 		{
-			if (target.Values == null)
-				yield break;
-
-			foreach (var targetValue in target.Values) {
+			foreach (var targetValue in target) {
```

Comment:
Nit: SelectMany might be more optimized than the generated state machine for foreach.